supervisor/src/bin/supervisor-encore.rs

```diff
@@ -58,16 +59,27 @@ pub async fn main() {
             )
             .expect("Failed to create process for gateways"),
         );
+        port += 1;
     }
 
-    let proxy = proxy::GatewayProxy::new(
-        reqwest::Client::new(),
-        std::net::SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), port),
-        service_ports.clone(),
-    );
-    let sv = Supervisor::new(procs);
+    let mut handles = Vec::with_capacity(2);
+    let mut results = Vec::with_capacity(handles.len());
```

Comment:
Isn't `handles.len()` always zero here? The capacity is 2 but length is 0. (Also seems like a highly unnecessary optimization ;))

Reply:
Haha, yeah. I might have had some extraterrestial help